### PR TITLE
feat(object): add piercy, a PC species intended for testing physics

### PR DESCRIPTION
**Pull request by anarchuser**
Depends on #29

### CODE REVIEW

File: src/model/model.cpp [hunk 1]
```diff
@@ -1,8 +1,43 @@
...
+
+double Point::operator / (const Point & p) const {return sqrt(pow(p.x - x, 2.0) + pow(p.y - y, 2.0));}
+
+Point Point::operator + (const Point & p) const {return Point(x + p.x, y + p.y);}
+Point Point::operator + (const Vector & v) const {return Point(x + v.length * cos(v.angle), y + v.length * sin(v.angle));}
+
+Point Point::operator - (const Point & p) const {return Point(x - p.x, y - p.y);}
+Point Point::operator - (const Vector & v) const {return * this + -v;}
+
+Vector Point::operator >= (const Point & p) const {
```
Comment (xeruf) [inline src/model/model.cpp:21]: add docs if you abuse operators

File: src/model/model.cpp [hunk 2]
```diff
@@ -1,8 +1,43 @@
...
+    const Point q(* this - p);
+    return Vector(atan(q.y / q.x), ORIGIN / q);
+}
+Vector Point::operator <= (const Point & p) const {return p >= * this;}
+
+/* Vector Math */
+Vector::Vector(double a, double l) : angle{(a < 0) ? a + M_PI : a}, length{fabs(l)} {}
+Vector::Vector(Point p) : Vector(atan(p.y / p.x), ORIGIN / p) {}
+Vector::Vector(int a, double l) : Vector(((double) a / 180 * M_PI), l) {}
+
```
Comment (xeruf) [inline src/model/model.cpp:31]: add a section comment here for Vector/Point compatibility

File: src/model/model.h [hunk 3]
```diff
@@ -3,15 +3,52 @@
...
+    Point (double x, double y);
+
+    Point & operator = (const Vector & v);
 
     bool operator == (const Point & p) const;
-    double operator / (const Point & p);
-    Point operator - (const Point & p);
-    Point operator + (const Point & p);
+
+    double operator / (const Point & p) const;      // Evaluates the distance between the two points
```
Comment (xeruf) [inline src/model/model.h]: make this docs

File: src/model/model.h [hunk 4]
```diff
@@ -3,15 +3,52 @@
...
+    double length;
+
+    explicit Vector (Point p);
+    Vector (double angle, double length);
+    Vector (int angle, double length);
+
+    Vector & operator = (const Point & p);
+
+    bool operator == (const Vector & v) const;
+    bool operator == (const Point & p) const;       // Compares 'this' with ORIGIN >> p
```
Comment (xeruf) [inline src/model/model.h]: this too

File: src/model/object/entity/individual/fred/fred.h [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#ifndef EVE_FRED_H
+#ifndef EVE_PIERCY_H
```
Comment (xeruf) [inline src/model/object/entity/individual/fred/fred.h]: undo dis

**Review by xeruf — CHANGES_REQUESTED** [review]

File: src/model/object/entity/individual/fred/fred.h (hunk 5 above)
Comment (anarchuser) [reply src/model/object/entity/individual/fred/fred.h]: undone

File: src/model/model.cpp (hunk 1 above)
Comment (anarchuser) [reply src/model/model.cpp:21]: resolved

File: src/model/model.cpp (hunk 2 above)
Comment (anarchuser) [reply src/model/model.cpp:31]: will come in a future update

File: src/model/model.h (hunk 3 above)
Comment (anarchuser) [reply src/model/model.h]: resolved

File: src/model/model.h (hunk 4 above)
Comment (anarchuser) [reply src/model/model.h]: resolved

**Review by xeruf — APPROVED** [review]

**Pull request merged**